### ISSUE TITLE
Extending with distributed rulesets

### ISSUE THREAD

**Issue by 50rayn** [daveshanley/vacuum#254]
Hello @daveshanley , it's me again 😄. Having some unclear things about Rulesets, especially [Extending Rulesets](https://meta.stoplight.io/docs/spectral/83527ef2dd8c0-extending-rulesets). I have read whole documentation on vacuum website, but I didn't find any information about extending with a [shared ruleset](https://meta.stoplight.io/docs/spectral/7895ff1196448-sharing-and-distributing-rulesets),  in my case File System option. 

For example, I have _openapi.yaml_, _main.ruleset.yaml_ and a folder _rules_ with custom rulesets separated by "category". The question is, how can I include the rulesets from _rules_ folder in _main.ruleset.yaml_ ? I have 71 additional custom rules which a are with the original Spectral, it would be sad to miss them 😕

_Folder structure_
```
|-- api
|---- rules
|---- stringTypeConstraints.yaml
|---- integerAndNumberTypeConstraints
|-- main.ruleset.yaml
|-- openapi.yaml
```

_main.ruleset.yaml_
```yaml
extends:
  - 'spectral:oas'
  - ./rules/stringTypeConstraints.yaml
  - ./rules/integerAndNumberTypeConstraints.yaml
  - ...
```

_stringTypeConstraints.yaml_
```yaml
rules:
  string-constraints:
    description: String constraints
    message: '{{error}}'
    severity: error
    type: style
    given: $..*[?(@.type === 'string')]
    then:
      function: schema
      functionOptions:
        schema:
          anyOf:
            - type: object
              required:
                - enum
              properties:
                enum:
                  type: array
            - type: object
              required:
                - maxLength
              properties:
                maxLength:
                  type: integer
            - type: object
              required:
                - minLength
              properties:
                minLength:
                  type: integer
            - type: object
              required:
                - pattern
              properties:
                pattern:
                  type: string
...
```
_integerAndNumberTypeConstraints.yaml_
```yaml
rules:
  integer-and-number-constraints:
    description: Integer and Number  constraints
    message: '{{error}}'
    severity: error
    type: style
    given: $..*[?(@.type === 'integer' || @.type === 'number')]
    then:
      function: schema
      functionOptions:
        schema:
          anyOf:
            - type: object
              required:
                - enum
              properties:
                enum:
                  type: array
            - type: object
              required:
                - maximum
              properties:
                maxLength:
                  type: integer
            - type: object
              required:
                - minimum
              properties:
                minLength:
                  type: integer
            - type: object
              required:
                - format
              properties:
                format:
                  type: string
...
```

**Comment by daveshanley** [comment]
Hi there, 

So this feature does not yet exist in vacuum, the ability to include references to other configs.

It can be added, for sure - but there is just no code in the tool to handle this capability right now. I'll add this as a feature and add it to the backlog.

**Comment by 50rayn** [comment]
Thanks in advance @daveshanley .
Will wait for news.

**Comment by ntotten** [comment]
I'd love to see this feature as well. While we can accomplish much of this by writing custom functions, the bar of doing that is a lot higher bar as you need to build a new `.so` for different Go version releases, etc.

**Comment by ankorstore-haddowg** [comment]
I have a shared ruleset distributed as an npm package i would love to be able to use with this, however that ruleset also itself extends from local js and yaml ruleset definitions any chance we could get vacuum to parse and use the js rulesets correctly too 🥺

**Comment by daveshanley** [comment]
I don't know what a JS ruleset looks like. Can you give me an example? I don't use Spectral myself anymore.

**Comment by daveshanley** [comment]
This feature is something I am planning on working on very soon, so the more requests and details in this ticket, the better.

**Comment by daveshanley** [comment]
This feature was implemented in `v0.6.0`

https://quobix.com/vacuum/rulesets/sharing/

**Comment by jjwedemyer** [comment]
Hi, does the release of `v0.6.0` imply that this statement is obsolete? https://quobix.com/vacuum/rulesets/understanding/#fn:1
